### PR TITLE
[Issue #2472] Terraform in ECS

### PR DESCRIPTION
**Pull request by coilysiren**
## Summary

Contributes to, but does not solve, #2472

## Changes proposed

Adds a new ECS service that contains:

- bin folder
- infra folder
- Makefile

I plan on using this ECS service to deploy terraform from inside of our VPC

## Context for reviewers

95% of this PR is copy paste, which is just how it is.

The parts that are specific to this PR are:

- `.dockerignore`
- `ecs-terraform/Dockerfile`
- `ecs-terraform/Makefile`

### CODE REVIEW

File: .dockerignore [hunk 1]
```diff
@@ -0,0 +1,3 @@
+node_modules
+**/.terraform/**
+.git
```
Comment (coilysiren) [inline .dockerignore:3]: The build was pulling in all the .terraform files, which made the build huge. Like 10 GB

File: ecs-terraform/Dockerfile [hunk 2]
```diff
@@ -0,0 +1,19 @@
+FROM hashicorp/terraform:1.9.7 AS base
+
+RUN mkdir -p /app
+WORKDIR /app
+ENTRYPOINT [ "sh", "-c" ]
+
+COPY --from=top-level-directory bin /app/bin
+COPY --from=top-level-directory infra /app/infra
+COPY --from=top-level-directory Makefile /app/Makefile
```
Comment (coilysiren) [inline ecs-terraform/Dockerfile:9]: Fancy new docker features here, ability to pull from a context above your current folder

File: ecs-terraform/Makefile [hunk 3]
```diff
@@ -0,0 +1,47 @@
+#############
```
Comment (coilysiren) [inline ecs-terraform/Makefile:1]: copy paste

File: infra/ecs-terraform/app-config/dev.tf [hunk 4]
```diff
@@ -0,0 +1,7 @@
+module "dev_config" {
```
Comment (coilysiren) [inline infra/ecs-terraform/app-config/dev.tf:1]: copy paste and snipped

File: infra/ecs-terraform/app-config/staging.tf [hunk 5]
```diff
@@ -0,0 +1,7 @@
+module "staging_config" {
```
Comment (coilysiren) [inline infra/ecs-terraform/app-config/staging.tf:1]: copy paste and snipped

File: infra/ecs-terraform/app-config/prod.tf [hunk 6]
```diff
@@ -0,0 +1,7 @@
+module "prod_config" {
```
Comment (coilysiren) [inline infra/ecs-terraform/app-config/prod.tf:1]: copy paste and snipped

File: infra/ecs-terraform/app-config/outputs.tf [hunk 7]
```diff
@@ -0,0 +1,31 @@
+output "app_name" {
```
Comment (coilysiren) [inline infra/ecs-terraform/app-config/outputs.tf:1]: copy paste

File: infra/ecs-terraform/app-config/main.tf [hunk 8]
```diff
@@ -0,0 +1,60 @@
+locals {
```
Comment (coilysiren) [inline infra/ecs-terraform/app-config/main.tf:1]: copy paste with slight modifications

File: infra/ecs-terraform/app-config/env-config/variables.tf [hunk 9]
```diff
@@ -0,0 +1,22 @@
+variable "app_name" {
```
Comment (coilysiren) [inline infra/ecs-terraform/app-config/env-config/variables.tf:1]: copy paste and snipped

File: infra/ecs-terraform/app-config/env-config/outputs.tf [hunk 10]
```diff
@@ -0,0 +1,11 @@
+output "service_config" {
```
Comment (coilysiren) [inline infra/ecs-terraform/app-config/env-config/outputs.tf:1]: copy paste and snipped

File: infra/ecs-terraform/app-config/env-config/environment-variables.tf [hunk 11]
```diff
@@ -0,0 +1,17 @@
+locals {
```
Comment (coilysiren) [inline infra/ecs-terraform/app-config/env-config/environment-variables.tf:1]: copy paste

File: infra/ecs-terraform/build-repository/main.tf [hunk 12]
```diff
@@ -0,0 +1,59 @@
+data "aws_iam_role" "github_actions" {
```
Comment (coilysiren) [inline infra/ecs-terraform/build-repository/main.tf:1]: copy paste

File: infra/ecs-terraform/service/main.tf [hunk 13]
```diff
@@ -0,0 +1,107 @@
+# docs: https://registry.terraform.io/providers/hashicorp/aws/latest/docs/resources/vpc
```
Comment (coilysiren) [inline infra/ecs-terraform/service/main.tf:1]: copy paste from the analytics module, which also is "triggered only" (eg. no web server)

File: infra/ecs-terraform/service/outputs.tf [hunk 14]
```diff
@@ -0,0 +1,24 @@
+output "service_endpoint" {
```
Comment (coilysiren) [inline infra/ecs-terraform/service/outputs.tf:1]: copy paste

File: infra/ecs-terraform/service/secrets.tf [hunk 15]
```diff
@@ -0,0 +1,16 @@
+module "secrets" {
```
Comment (coilysiren) [inline infra/ecs-terraform/service/secrets.tf:1]: copy paste

File: ecs-terraform/Makefile [hunk 16]
```diff
@@ -0,0 +1,47 @@
...
+
+##################
+# Build Commands #
+##################
+
+build:
+	docker buildx build \
+		--build-context top-level-directory=../ \
+		--tag $(notdir $(shell pwd)):latest \
+		.
```
Comment (coilysiren) [inline ecs-terraform/Makefile:37]: fancy

**Review by mdragon — APPROVED** [review]
3 files you had us focus on look good.

**Pull request merged**